### ISSUE TITLE
`base_ring` vs `coefficient_ring`: methods and documentation

### ISSUE THREAD

**Issue by matthias314** [Nemocas/AbstractAlgebra.jl#1207]
`AbstractAlgebra` provides the functions `base_ring` and `coefficient_ring`. Sometimes they are both defined and synonymous:
```
julia> R, x = ZZ['x']; base_ring(R), coefficient_ring(R)
(Integers, Integers)
```
For types different from rings and algebras, however, only `base_ring` seems to be defined. Example:
```
julia> M = FreeModule(ZZ, 3); base_ring(M)
Integers

julia> coefficient_ring(M)
ERROR: MethodError: no method matching coefficient_ring(::AbstractAlgebra.Generic.FreeModule{BigInt})
```
For me, `coefficient_ring` would be the more intuitive terminology in this case. Is this intended behavior?

Also, the documentation for `base_ring` is very short. The function is mentioned on the page [Ring Interface](https://nemocas.github.io/AbstractAlgebra.jl/dev/ring_interface/#Ring-Interface), but for instance not on the page [Module Interface](https://nemocas.github.io/AbstractAlgebra.jl/dev/module_interface/#Module-Interface).  For `coefficient_ring` I couldn't find any documentation.

**Comment by thofma** [comment]
I agree that we should add it. I started a discussion over at https://github.com/oscar-system/Oscar.jl/issues/1505, since it has to be coordinated with some other packages.
